gnd/src/main/java/com/google/android/gnd/rx/Result.java

```diff
@@ -54,4 +57,35 @@ protected Result(@Nullable T value, @Nullable Throwable error) {
   public String toString() {
     return error().map(t -> "Error: " + t).orElse("Value: " + value);
   }
+
+  /**
+   * Returns a {@link Flowable} that maps values emitted from the source stream as Result instances.
+   * Errors in the provided stream are handled and wrapped in a Result with ERROR state. The
+   * returned stream itself should never fail with an error.
+   *
+   * @param source the stream to be modified.
+   * @param <T> the type of value being emitted.
+   */
+  public static <T> Flowable<Result<T>> wrap(Flowable<T> source) {
```

Comment:
Would changing these to operate on `Single` instead of `Flowable` possibly make more sense semantically?

Reply:
I tried this out and it seems to be more intuitive with Single, but in the end, wrap/unwrap aren't needed for this PR so removing them. Stashing them in [this gist](https://gist.github.com/gino-m/efcfd158a4d6e4461ac17a1a064d8a7e) for use when we want to wrap potentials errors without breaking a `Flowable` or `Observable`.

Reply:
I don't have anything useful to add to the conversation. @scolsen ?

Reply:
There's a long thread on #84, but I think we can leave that for a separate PR since this one is already growing unwieldy (sorry for that!).

Reply:
Thanks! Dealing with it separately sounds ok to me